app/src/sharedTest/java/org/oppia/android/app/policies/PoliciesFragmentTest.kt

```diff
@@ -325,6 +329,44 @@ class PoliciesFragmentTest {
     }
   }
 
+  @Test
+  fun testPoliciesFragment_fragmentLoaded_verifyCorrectArgumentsPassed() {
+    launch<PoliciesFragmentTestActivity>(
+      createPoliciesFragmentTestIntent(
+        getApplicationContext(),
+        PolicyPage.TERMS_OF_SERVICE
+      )
+    ).use { scenario ->
+      testCoroutineDispatchers.runCurrent()
+      scenario.onActivity { activity ->
+
+        val policiesFragment = activity.supportFragmentManager
+          .findFragmentById(R.id.policies_fragment_placeholder) as PoliciesFragment
+
+        val policiesActivityParams = activity.intent.getProtoExtra(
+          PoliciesFragmentTestActivity.POLICIES_FRAGMENT_TEST_POLICY_PAGE_PARAMS_PROTO,
+          PoliciesActivityParams.getDefaultInstance()
+        )
+        val policiesFragmentArguments =
+          PoliciesFragmentArguments
+            .newBuilder()
+            .setPolicyPage(policiesActivityParams.policyPage)
+            .build()
+
+        val args = checkNotNull(policiesFragment.arguments) {
+          "Expected arguments to be passed to PoliciesFragment"
+        }
+        val receivedPolicies =
+          args.getProto(
+            POLICIES_FRAGMENT_POLICY_PAGE_ARGUMENT_PROTO,
+            PoliciesFragmentArguments.getDefaultInstance()
+          )
+
+        assertThat(receivedPolicies).isEqualTo(policiesFragmentArguments)
```

Comment:
Prefer checking the properties of the proto rather than the whole thing (since it helps make the test more deliberate and contextual).